piece.cc

```diff
@@ -39,11 +39,12 @@ int Piece::move(Piece *currentTile, Piece *targetTile, int newX, int newY) {
 
     */
 
-    updateLegalMoves();
+    std::map<Box, int> tempLegalMovesMap = updateLegalMoves();
+    legalMovesMap = &tempLegalMovesMap;
```

Comment:
Does this work? I thought this was the thing that was giving you trouble lol.

Reply:
idk apparently this works I wasn't able to do
`legalMovesMap = &updateLegalMoves()`
also apparently map has an operator= and a copy constructor